extensions/functions_arithmetic_decimal.yaml

```diff
@@ -174,3 +174,95 @@ aggregate_functions:
         decomposable: MANY
         intermediate: "DECIMAL<38,S>"
         return: "DECIMAL<38,S>"
+
+  -
+    name: "power"
+    description: "Take the power with x as the base and y as exponent."
+    impls:
+      - args:
+          - name: x
+            value: "DECIMAL<P,S>"
+          - name: y
+            value: "DECIMAL<P,S>"
+        options:
+          overflow:
+            values: [ SILENT, SATURATE, ERROR ]
+        return: fp64
+
+  -
+    name: "sqrt"
+    description: "Square root of the value"
+    impls:
+      - args:
+          - name: x
+            value: "DECIMAL<P,S>"
+        options:
+          rounding:
+            values: [ TIE_TO_EVEN, TIE_AWAY_FROM_ZERO, TRUNCATE, CEILING, FLOOR ]
+          on_domain_error:
+            values: [ NAN, ERROR ]
+        return: fp64
+
+  -
+    name: "factorial"
+    description: >
+      Return the factorial of a given integer input.
+      The factorial of 0! is 1 by convention.
+      Negative inputs will raise an error.
+    impls:
+      - args:
+          - value: "DECIMAL<P,S>"
+            name: "n"
+        options:
+          overflow:
+            values: [ SILENT, SATURATE, ERROR ]
+        return: "DECIMAL<P,S>"
```

Comment:
Will the precision and scale stay the same as a result of this operation or do we need to extend it?  I suppose overflow is quickly likely so it's not a big deal if we don't rescale.

Reply:
max it will be Decimal(37, S), which  supports factorial of max 33 (verified in duckdb and Snowflake). In both these engines type of return datatype was max size and didn't depend on input size. So I made it Decimal(38, S)